test-data/ReactThisBinding/ReactThisBindingIssueWithDecorator-passing.tsx

```diff
@@ -0,0 +1,30 @@
+import React = require('react');
+
+export class BaseClass extends React.Component<{}, {}> {
+    public notALocalMethod(): void {
+    }
+}
+declare const autobind: MethodDecorator;
+
+export class MyComponent extends BaseClass {
+
+    @autobind
+    private listener1(): void {
+    }
+
+    @autobind
+    private listener2(): void {
+    }
+
+    // this listener is a lambda, so it has its 'this' reference bound correctly.
+    private listener3: () => void = (): void => {};
+
+    public render() {
+        return <div
+            onClick={this.listener1}>
+            onMouseOver={this.listener2}>
+            onMouseDown={this.listener3}>
```

Comment:
You've got a few syntax errors here in the test: `>` should only come after the last attribute.

Reply:
Ahh, my bad, I copied from existing one and I didn't notice that for some reason :smile: , I'll fix the original file as well :)